highlighters/main/mh-parse.zsh

```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
+
+#
+# This file runs main highlighter on a specified file
+# i.e. parses the file with the highlighter. Outputs
+# running time (stderr) and resulting region_highlight
+# (stdout).
+#
+
+# Strive for -f
+# .. occuring problems (no q flag), but at least try to quote $0 and $@
+[[ -z "$ZSH_VERSION" ]] && exec /usr/bin/env zsh -f -c "source \"$0\" \"$1\" \"$2\" \"$3\""
```

Comment:
About quoting:   `/usr/bin/env zsh -fc 'source "$@"' - "$0" "$@"` should do the trick.